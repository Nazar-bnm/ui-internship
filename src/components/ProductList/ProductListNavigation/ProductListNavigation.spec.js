import React from 'react';
import { shallow } from 'enzyme';
import configureStore from 'redux-mock-store';
import toJson from 'enzyme-to-json';
import ItemListContainer from '../ItemsList/ItemsListContainer';

const configureMockStore = configureStore();

const initialState = {
  ascendingOrder: true,
  orderType: 'Position',
  chosenItemsOnPage: 6,
  itemList: [],
  error: null,
}

const products = [{dress: 1}];
const errorMessage = 'Error happened';
const store = configureMockStore({reducerProductList: initialState});

it('should receive the state from the store', () => {
  const wrapper = shallow(
      <ItemListContainer store={ store } />
  ).dive();
  expect(toJson(wrapper)).toMatchSnapshot();  
});

it('should path the proper array of products to the action', () => {
  const wrapper = shallow(
    <ItemListContainer store={ store } />
  ).dive();
  expect(wrapper.props().onGetProductsSuccess(products).products).toBe(products);
});

it('should path an error message to the action', () => {
  const wrapper = shallow(
    <ItemListContainer store={ store } />
  ).dive();
  expect(wrapper.props().onGetProductsError(errorMessage).error).toBe(errorMessage);
});