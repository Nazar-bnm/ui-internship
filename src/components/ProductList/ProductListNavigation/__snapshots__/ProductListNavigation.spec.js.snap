// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProductListNavigation /> should match snapshot 1`] = `
<div
  className="product-list__dropdowns-wrapper"
>
  <div
    className="product-list__dropdowns-labels-wrapper"
  >
    <span
      className="product-list__dropdowns-labels"
    >
      sort by
    </span>
    <Dropdown
      className=""
      menuOptions={
        Array [
          Object {
            "label": "Position",
            "value": "Position",
          },
          Object {
            "label": "Price",
            "value": "Price",
          },
          Object {
            "label": "Name",
            "value": "Name",
          },
        ]
      }
      onDropdownChange={[Function]}
      placeholder=""
      value="Position"
    />
    <Button
      customClass="product-list__arrow-btn"
      icon="arrow down"
      onClick={[Function]}
      onClickFunction={[MockFunction]}
    />
  </div>
  <div
    className="product-list__dropdowns-labels-wrapper"
  >
    <span
      className="product-list__dropdowns-labels"
    >
      items on page
    </span>
    <Dropdown
      className=""
      menuOptions={
        Array [
          Object {
            "label": "6",
            "value": "6",
          },
          Object {
            "label": "9",
            "value": "9",
          },
          Object {
            "label": "12",
            "value": "12",
          },
        ]
      }
      onDropdownChange={[Function]}
      placeholder=""
      value="6"
    />
  </div>
</div>
`;
